src/queries/SUE/areaService.tsx

```diff
@@ -17,7 +19,21 @@ export const areaService = async () => {
     }
   };
 
-  return await (
-    await fetch(`${serverUrl}/${id}?selectAttributes=postalCodes`, areaServiceFetchObj)
+  const getRS = await (
+    await fetch(`${serverUrl}/${id}?selectAttributes=rs`, areaServiceFetchObj)
+  ).json();
+
+  const postalCodes = await (
+    await fetch(
+      `${serverUrl}/getByRs?rs=${getRS?.rs}*&selectAttributes=postalCodes`,
+      areaServiceFetchObj
+    )
   ).json();
+
+  return {
+    postalCodes: _uniqBy(
+      postalCodes?.values?.flatMap((item: any) => item.postalCodes || []),
+      (code) => code
+    )
```

Comment:
lodashs _uniq(Array) should be enough here

Reply:
But we don't have just one array. We have an array of different zip codes inside an array. So I'm not sure if it will work correctly with just _uniq(Array). The array we use uniq on here looks like the one below.

here I think that for `_uniqBy` to work correctly we need to give an array with only postal codes in it. For example= [2312,3123,42323...]. For this reason we use `flatMap` to format `postalCodes.values` as an array containing only postal codes.

```
{
  "values": [
    {},
    {
      "postalCodes": [
        "39539",
        "39615"
      ]
    },
    {
      "postalCodes": [
        "39596"
      ]
    },
    {
      "postalCodes": [
        "39596"
      ]
    },
    {
      "postalCodes": [
        "39606"
      ]
    },
    {
      "postalCodes": [
        "39596"
      ]
    },
    {
      "postalCodes": [
        "39596"
      ]
    },
    {
      "postalCodes": [
        "39579"
      ]
    },
    {
      "postalCodes": [
        "39596"
      ]
    }
  ],
  "offset": 0,
  "totalCount": 9,
  "count": 9
}
```

Reply:
sure, flat map is needed. but uniqBy shuld be able to be changed to uniq

Reply:
Ow mann! I didn't understand what you meant by that. I'm sorry. I will learn the difference between uniyBy and uniq. 👍

Reply:
https://github.com/smart-village-solutions/smart-village-app-app/commit/9ee1866a9a7aae19d1adf5afa86c0aae73a082ac